libs/telemetry/logger.go

```diff
@@ -58,16 +73,9 @@ func Flush(ctx context.Context, apiClient DatabricksApiClient) {
 	defer cancel()
 	l := fromContext(ctx)
 
-	// We pass the API client as an arg to mock it in unit tests.
-	if apiClient == nil {
-		var err error
-
-		// Create API client to make the the telemetry API call.
-		apiClient, err = client.New(root.WorkspaceClient(ctx).Config)
```

Comment:
Changed the function signature to always take in an apiClient since `root.WorkspaceClient` will not be configured for all CLI commands as we add support for more commands.